File: src/services/timezone-spoof.ts
/**
 * 时区/NTP 伪装服务
 *
 * 提供时区伪装配置的 TS 类型和 Tauri 命令绑定
 */

import { invoke } from '@tauri-apps/api/core'

// ── 类型 ──────────────────────────────────────────────────────────

/** NTP 服务器选择策略 */
export type NtpStrategy = 'Auto' | 'Manual' | 'Disabled'

/** 时区/NTP 伪装配置 */
export interface TimezoneSpoofConfig {
  enabled: boolean
  ntp_strategy: NtpStrategy
  manual_ntp_server?: string
  ntp_interval_min: number
  write_to_system: boolean
  dialer_proxy?: string
}

// ── 命令 ──────────────────────────────────────────────────────────

/** 获取时区伪装配置 */
export async function timezoneSpoofGetConfig(): Promise<TimezoneSpoofConfig> {
  return invoke('timezone_spoof_get_config')
}

/** 更新时区伪装配置 */
export async function timezoneSpoofUpdateConfig(
  config: TimezoneSpoofConfig
): Promise<void> {
  return invoke('timezone_spoof_update_config', { config })
}

/** 根据国家代码获取推荐的 NTP 服务器 */
export async function timezoneSpoofGetNtpServer(
  countryCode: string
): Promise<string> {
  return invoke('timezone_spoof_get_ntp_server', { countryCode })
}

/** 根据国家代码获取时区 */
export async function timezoneSpoofGetTimezone(
  countryCode: string
): Promise<string> {
  return invoke('timezone_spoof_get_timezone', { countryCode })
}

/** 根据时区获取 locale */
export async function timezoneSpoofGetLocale(
  timezone: string
): Promise<string> {
  return invoke('timezone_spoof_get_locale', { timezone })
}

File: src-tauri/src/cmd/timezone_spoof.rs
use super::{CmdResult, StringifyErr};
/**
 * 时区/NTP 伪装 Tauri 命令
 */
use crate::core::timezone_spoof::TimezoneSpoofConfig;

/// 获取时区伪装配置
#[tauri::command]
pub fn timezone_spoof_get_config() -> CmdResult<TimezoneSpoofConfig> {
    Ok(crate::core::coordinator::get_coordinator()
        .get_advanced_config()
        .timezone_spoof)
}

/// 更新时区伪装配置
#[tauri::command]
pub async fn timezone_spoof_update_config(config: TimezoneSpoofConfig) -> CmdResult<()> {
    crate::core::coordinator::update_advanced_config(move |advanced| {
        advanced.timezone_spoof = config;
    })
        .await
        .stringify_err()
}

/// 根据国家代码获取推荐的 NTP 服务器
#[tauri::command]
pub fn timezone_spoof_get_ntp_server(country_code: String) -> CmdResult<String> {
    Ok(crate::core::timezone_spoof::select_ntp_server(&country_code))
}

/// 根据国家代码获取时区
#[tauri::command]
pub fn timezone_spoof_get_timezone(country_code: String) -> CmdResult<String> {
    Ok(crate::core::timezone_spoof::country_to_timezone(&country_code).to_string())
}

/// 根据时区获取 locale
#[tauri::command]
pub fn timezone_spoof_get_locale(timezone: String) -> CmdResult<String> {
    Ok(crate::core::timezone_spoof::timezone_to_locale(&timezone))
}
